app/src/test/java/com/chesire/nekome/ActivityViewModelTests.kt

```diff
@@ -77,38 +85,45 @@ class ActivityViewModelTests {
         val classUnderTest = ActivityViewModel(
             mockAccessTokenRepository,
             mockLogoutHandler,
+            settings,
             testDispatcher,
             mockUserRepository
         )
 
-        classUnderTest.logout { }
+        classUnderTest.logout()
 
         verify { mockLogoutHandler.executeLogout() }
     }
 
+
     @Test
-    fun `logout executes callback after logoutHandler`() {
-        val mockAccessTokenRepository = mockk<AccessTokenRepository>()
+    fun `if user is not logged it then details screen navigation event is emited`() {
+        val mockAccessTokenRepository = mockk<AccessTokenRepository>() {
+            every { accessToken } returns ""
+        }
         val mockLogoutHandler = mockk<LogoutHandler> {
             every { executeLogout() } just Runs
         }
         val mockUserRepository = mockk<UserRepository> {
             every { user } returns mockk()
         }
-        val mockCallback = mockk<() -> Unit>(relaxed = true)
 
         val classUnderTest = ActivityViewModel(
             mockAccessTokenRepository,
             mockLogoutHandler,
+            settings,
             testDispatcher,
             mockUserRepository
         )
 
-        classUnderTest.logout(mockCallback)
+        val mockObserver = mockk<Observer<NavDirections>> {
+            every { onChanged(any()) } just Runs
+        }
+
+        classUnderTest.navigation.observeForever(mockObserver)
```

Comment:
> check the contents of the observer rather than a verify.

man this words changed the way I will test the bservers forever :D 

so i have used `.value` and checked them directly and worked on the first try!

Reply:
Nice, glad it could help!